src/CellularPanel.cpp

```diff
@@ -962,29 +961,30 @@ void CellularPanel::HandleClick( const TrackPanelMouseEvent &tpmEvent )
          HandleMotion( tpmState );
       }
       ProcessUIHandleResult(
-         pCell.get(), pCell.get(), refreshResult);
+         pCell, pCell, refreshResult);
       state.mMouseOverUpdateFlags |= refreshResult;
    }
 }
 
-void CellularPanel::DoContextMenu( TrackPanelCell *pCell )
+void CellularPanel::DoContextMenu( const std::shared_ptr<TrackPanelCell> &pCell )
 {
-   if( !pCell ) {
-      pCell = GetFocusedCell();
-      if( !pCell )
+   std::shared_ptr<TrackPanelCell>pCellLocal = pCell;
```

Comment:
So this variable initialization might disappear -- but even if this were needed, I would also use `auto` to simplify the declaration, avoiding the the verbosity of the long type name.  Let the compiler deduce it instead.

Reply:
In general I don't like "auto" because it makes the code less comprehensible, for me anyway.  But like you also suggest, this initialization should disappear after I make the other change.